### PR TITLE
Onclick

### PR DESCRIPTION
**Pull request by amusinamaria**
Make MovieCard parcelable and pass it to DetailsFragment.
Add ViewBinding

### CODE REVIEW

**Review by nikslushkin — APPROVED** [review]

File: app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt [hunk 1]
```diff
@@ -2,15 +2,21 @@ package com.github.amusinamaria.ui
...
+
+    private lateinit var binding: ActivityMainBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
 
         if (savedInstanceState == null) {
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt]: Use savedInstanceState ?:  supportFragmentManager.beginTransaction()
This is Kotlin style

File: app/src/main/java/com/github/amusinamaria/ui/details/CastAdapter.kt [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
 import com.bumptech.glide.load.MultiTransformation
 import com.bumptech.glide.load.resource.bitmap.CenterCrop
 import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
 import com.bumptech.glide.request.RequestOptions
 import com.github.amusinamaria.R
+import com.github.amusinamaria.databinding.CastCardBinding
 import com.github.amusinamaria.repository.CastCard
 import jp.wasabeef.glide.transformations.RoundedCornersTransformation
 
+
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/github/amusinamaria/ui/details/CastAdapter.kt]: Remove

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt [hunk 3]
```diff
@@ -4,38 +4,46 @@ import android.os.Bundle
...
-import com.github.amusinamaria.R
-import com.github.amusinamaria.repository.MockDataSource
+import com.github.amusinamaria.databinding.FragmentDetailsBinding
+import com.github.amusinamaria.repository.MovieCard
+import com.github.amusinamaria.ui.MainActivity.Companion.MOVIE_ARGS_KEY
 
 class DetailsFragment : Fragment() {
 
-    private lateinit var castRecycler: RecyclerView
+    private var _binding: FragmentDetailsBinding? = null
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt:15]: In onDestroyView, you must set this field to null.

Comment (Pavel-Sidorovich) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt:15]: In onDestroyView, you must set this field to null.

File: app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt [hunk 4]
```diff
@@ -21,12 +27,22 @@ class MainActivity : AppCompatActivity(), MoviesListFragment.MovieClickListener
         }
     }
 
-    override fun onClickShowMovieDetails() {
+    fun showMovieDetails(movieCard: MovieCard) {
+        val detailsFragment = DetailsFragment()
+        val args = Bundle()
+        args.putParcelable(MOVIE_ARGS_KEY, movieCard)
+        detailsFragment.arguments = args
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt]: Use DetailsFragment.newInstance(movieCard) and put newInstance fun to DetailsFragment companion object

File: app/src/main/java/com/github/amusinamaria/ui/list/MoviesListFragment.kt [hunk 5]
```diff
@@ -1,52 +1,41 @@
...
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Button
 import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.GridLayoutManager
-import androidx.recyclerview.widget.RecyclerView
 import com.github.amusinamaria.R
+import com.github.amusinamaria.databinding.FragmentMoviesListBinding
 import com.github.amusinamaria.repository.MockDataSource
+import com.github.amusinamaria.ui.MainActivity
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/list/MoviesListFragment.kt]: remove it

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt [hunk 6]
```diff
@@ -4,38 +4,46 @@ import android.os.Bundle
...
+    ): View {
+        _binding = FragmentDetailsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<ImageView>(R.id.backArrow).setOnClickListener {
+        movieCard = arguments?.getParcelable(MOVIE_ARGS_KEY)!!
+        val movieTitle = binding.detailsTitle
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: Why you need that val?

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt [hunk 7]
```diff
@@ -4,38 +4,46 @@ import android.os.Bundle
...
-        castRecycler.apply {
+        binding.castRecycler.apply {
             layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
             setHasFixedSize(true)
         }
         castAdapter = CastAdapter().apply {
             setHasStableIds(true)
         }
-        castRecycler.adapter = castAdapter
+        binding.castRecycler.adapter = castAdapter
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: try to use binding.apply {}. for reducing the count of "binding."

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt [hunk 8]
```diff
@@ -45,7 +53,7 @@ class DetailsFragment : Fragment() {
 
     private fun updateCastData() {
         castAdapter.apply {
-            bindCastCards(MockDataSource().getCastCards())
+            bindCastCards(movieCard.cast)
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: Do you really need to use APPLY for one function call?

**Review by n-balashov — APPROVED** [review]

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt (hunk 6 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt (hunk 7 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt (hunk 8 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt (hunk 3 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt:15]: Done

Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt:15]: Done

File: app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt (hunk 1 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt (hunk 4 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/details/CastAdapter.kt (hunk 2 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/details/CastAdapter.kt]: Done

File: app/src/main/java/com/github/amusinamaria/ui/list/MoviesListFragment.kt (hunk 5 above)
Comment (amusinamaria) [reply app/src/main/java/com/github/amusinamaria/ui/list/MoviesListFragment.kt]: Done

**Review by nikslushkin — APPROVED** [review]

File: app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt [hunk 9]
```diff
@@ -2,31 +2,36 @@ package com.github.amusinamaria.ui
...
+            .apply {
+                add(R.id.fragments_container, MoviesListFragment())
+                commit()
+            }
     }
 
-    override fun onClickShowMovieDetails() {
+    fun showMovieDetails(movieCard: MovieCard) {
         supportFragmentManager.beginTransaction()
+            .setTransition(TRANSIT_FRAGMENT_FADE)
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt:30]: Why not inside Apply?

File: app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt [hunk 10]
```diff
@@ -4,38 +4,45 @@ import android.os.Bundle
...
+        binding.apply {
+            detailsTitle.text = movieCard.title
+            backArrow.setOnClickListener {
+                fragmentManager?.popBackStack()
+            }
+            castRecycler.apply {
+                layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
+                setHasFixedSize(true)
+            }
+            castRecycler.adapter = castAdapter
```
Comment (n-balashov) [inline app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt:44]: You can move it inside Apply 2 lines upper

**Review by n-balashov — APPROVED** [review]

**Review by Pavel-Sidorovich — APPROVED** [review]

**Pull request merged**